### ISSUE TITLE
Does not work with Android O

### ISSUE THREAD

**Issue by rekire** [bjoernQ/adc#1]
With Android O it always try to open: com.android.systemui.recents.RecentsActivity

**Comment by bjoernQ** [comment]
Works for me on Nexus 5X and emulator (latest O previews). Can you give more details?

**Comment by rekire** [comment]
Hey Björn,

I tied it on two Windows 10 computers with my Pixel with Android O (Build OPP4.170623.014). On my other computer I get a different Activity but nothing related with the app which is currently on the display.

I hope this helps:
```
sailfish:/ $ uiautomator dump /dev/tty
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?><hierarchy rotation="0"><node index="0" text="" resource-id="" class="android.widget.FrameLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="" class="android.widget.LinearLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="" class="android.widget.FrameLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="com.example.debug:id/action_bar_root" class="android.widget.LinearLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="android:id/content" class="android.widget.FrameLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="com.example.debug:id/drawer_layout" class="android.support.v4.widget.DrawerLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="" class="android.widget.RelativeLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,1794]"><node index="0" text="" resource-id="com.example.debug:id/toolbar_container" class="android.widget.LinearLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,63][1080,336]"><node index="0" text="" resource-id="com.example.debug:id/toolbar" class="android.view.ViewGroup" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,63][1080,210]"><node index="0" text="" resource-id="" class="android.widget.ImageButton" package="com.example.debug" content-desc="Nach oben" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,63][147,210]" /><node index="1" text="Example" resource-id="" class="android.widget.TextView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[189,101][628,172]" /><node index="2" text="" resource-id="" class="android.support.v7.widget.LinearLayoutCompat" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[828,63][1080,210]"><node index="0" text="" resource-id="com.example.debug:id/action_search" class="android.widget.TextView" package="com.example.debug" content-desc="Suche" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="true" password="false" selected="false" bounds="[828,73][954,199]" /><node index="1" text="" resource-id="com.example.debug:id/action_cart" class="android.widget.RelativeLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="true" password="false" selected="false" bounds="[954,73][1080,199]"><node index="0" text="" resource-id="" class="android.widget.ImageView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[985,104][1048,167]" /><node index="1" text="" resource-id="" class="android.widget.RelativeLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[1027,97][1069,139]"><node index="0" text="" resource-id="com.example.debug:id/red_bubble" class="android.widget.ImageView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[1027,97][1069,139]" /><node index="1" text="0" resource-id="com.example.debug:id/cart_item_count" class="android.widget.TextView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[1034,103][1062,132]" /></node></node></node></node><node index="1" text="" resource-id="com.example.debug:id/tab_layout" class="android.widget.HorizontalScrollView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,210][1080,336]"><node index="0" text="" resource-id="" class="android.widget.LinearLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,210][1080,336]"><node index="0" text="" resource-id="" class="android.support.v7.app.ActionBar$Tab" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="true" bounds="[0,210][540,336]"><node index="0" text="SORTIMENT" resource-id="" class="android.widget.TextView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="true" bounds="[169,247][371,298]" /></node><node index="1" text="" resource-id="" class="android.support.v7.app.ActionBar$Tab" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[540,210][1080,336]"><node index="0" text="FAVORITEN" resource-id="" class="android.widget.TextView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[714,247][906,298]" /></node></node></node></node><node index="1" text="" resource-id="com.example.debug:id/fragment_container" class="android.widget.FrameLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,336][1080,1794]"><node index="0" text="" resource-id="" class="android.widget.FrameLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,336][1080,1794]"><node index="0" text="" resource-id="com.example.debug:id/status_view" class="android.widget.ScrollView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,336][1080,1794]"><node index="0" text="" resource-id="" class="android.widget.RelativeLayout" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[198,516][881,1614]"><node index="0" text="" resource-id="com.example.debug:id/image" class="android.widget.ImageView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[342,558][736,952]" /><node index="1" text="Ups, da ist wohl etwas schief gegangenÔÇª" resource-id="com.example.debug:id/error_headline" class="android.widget.TextView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[198,1005][881,1163]" /><node index="2" text="Bitte versuchen Sie es sp├ñter noch einmal. Sollte das Problem weiterhin auftreten, kontakieren Sie bitte unseren Kundenservice." resource-id="com.example.debug:id/error_message" class="android.widget.TextView" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[198,1189][881,1393]" /><node index="3" text="ERNEUT VERSUCHEN" resource-id="com.example.debug:id/button" class="android.widget.Button" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="true" enabled="true" focusable="true" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[319,1446][760,1572]" /></node></node></node></node><node index="2" text="" resource-id="com.example.debug:id/toolbar_shadow" class="android.view.View" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,336][1080,347]" /></node></node></node></node></node></node><node index="1" text="" resource-id="android:id/statusBarBackground" class="android.view.View" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][1080,63]" /><node index="2" text="" resource-id="android:id/navigationBarBackground" class="android.view.View" package="com.example.debug" content-desc="" checkable="false" checked="false" clickable="false" enabled="true" focusable="false" focused="false" scrollable="false" long-clickable="false" password="false" selected="false" bounds="[0,0][0,0]" /></node></hierarchy>
```

[`dumpsys activity top`](https://github.com/bjoernQ/adc/files/1172110/dumpsys.activity.top.txt)

**Comment by bjoernQ** [comment]
Now I know that it's about the "GO TO ACTIVITY" option - it worked for me with the betas but stopped working with the final O release. I'll have a look and fix this with the next version.

**Comment by bjoernQ** [comment]
fixed in 1.0.3
